lib/IDE/SwiftSourceDocInfo.cpp

```diff
@@ -1247,7 +1261,19 @@ struct RangeResolver::Implementation {
     Decl *D = Node.is<Decl*>() ? Node.get<Decl*>() : nullptr;
     analyzeDecl(D);
     auto &DCInfo = getCurrentDC();
-    switch (getRangeMatchKind(Node.getSourceRange())) {
+
+    auto NodeRange = Node.getSourceRange();
+
+    // Widen the node's source range to include all attributes to get a range
+    // match if a function with its attributes has been selected.
+    if (auto D = Node.dyn_cast<Decl *>()) {
+      for (auto Attr : D->getAttrs()) {
+        if (Attr->getRange().isValid())
+          NodeRange = NodeRange.widened(Attr->getRange());
+      }
+    }
```

Comment:
Can we move this function as a member function of `Decl`? we can name it as "getRangeIncludingAttrs()".